lib/widgets/message_list.dart

```diff
@@ -460,6 +461,19 @@ class DmRecipientHeader extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    final String title;
+    if (message.allRecipientIds.length > 1) {
+      final otherNames = message.allRecipientIds
+        .where((id) => id != store.account.userId)
+        .map((id) => store.users[id]?.fullName ?? '(unknown user)')
+        .sorted()
+        .join(", ");
+      title = 'You and $otherNames';
+    } else {
+      title = 'You with yourself'; // TODO pick string; web has glitchy "You and $yourname"
+    }
```

Comment:
Instead of using a filtered `message.allRecipientIds`, we could put `DmNarrow.ofMessage(message, …)` in a variable (we already do one of those below, although that's in an `onTap` callback), and get `.otherRecipientIds` from that. If we think it's cleaner and not too much overhead for a build method.

Reply:
Yeah. It wouldn't be much if at all shorter that way ­— the `.where` only adds one line — and I think this logic is fairly self-explanatory to read.

Conversely, there would be some distinct extra overhead in the DmNarrow approach. The `DmNarrow.ofMessage` constructor would allocate a List to hold all the recipient IDs, then `otherRecipientIds` would allocate another, and then the `sorted` would allocate a third. In this version, I believe the only allocation is for `sorted`.

I think ideally we would have a single DmNarrow object per distinct DM narrow (and similarly perhaps TopicNarrow and StreamNarrow), and an efficient way to look up that single object from a message that's in it. Then those allocations would be done just once for a given narrow; and it'd give us a handy place to go further and memoize the sorted list of names, too (though that one is a bit trickier in that it needs to get invalidated when names change), and/or the string we get from those with `join`. At some point when we have a stronger need for it I may try to arrange something like that.

Reply:
Makes sense.